### ISSUE TITLE
Potentially incorrect order of operations and confusing logic

### ISSUE THREAD

**Issue by phzwart** [mlexchange/mlex_dlsia_segmentation_prototype#14]
https://github.com/mlexchange/mlex_dlsia_segmentation_prototype/blob/87539a90540cbefeab9db0c1675edfdb7ce7cc49/src/seg_utils.py#L342

I think it is better to split this function into separate chunks:

1. one overall function that get a single frame from tiled
2. ship the frame (or frames) to the segment function. in the segment function we run qlty internally, segment stuff and return back in image in the right shape.
3. push results back via tiled

I will write step 2 right now

**Comment by TibbersHao** [comment]
Yes @phzwart with the latest PR #18 the previous `segment` function has been replaced with the new function which takes a single frame and run `qlty` internally. Writing back to tiled is one line of code so it does not hurt too much to have part 3 with 2 to my opinion. 

Please let me know if you have additional concerns for this issue, and I am happy to address.
